### PR TITLE
Cncf namespace samples

### LINKED ISSUE #44813: Add sample demonstrating publishing CNCF CloudEvent to Event Grid namespace topic
We can publish the event using the protocol methods. Related https://github.com/Azure/azure-sdk-for-net/issues/43371

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/44813

### CODE REVIEW

File: sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/samples/Sample2_CNCF.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+```
+
+Next, we receive the events using the `EventGridReceiverClient`.
+
+```C# Snippet:ReceiveCNCFEvent
+var receiver = new EventGridReceiverClient(new Uri(namespaceTopicHost), topicName, subscriptionName, new AzureKeyCredential(namespaceKey));
+Response response = await receiver.ReceiveAsync(maxEvents: 1, maxWaitTime: TimeSpan.FromSeconds(10), new RequestContext());
+using var document = JsonDocument.Parse(response.Content);
+
+var eventResponse = document.RootElement.GetProperty("value").EnumerateArray().First();
```
Comment (JoshLove-msft) [inline sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/samples/Sample2_CNCF.md:26]: I tried using ToDynamicJson instead, but it doesn't seem to allow you to let you convert to bytes, which I would need to pass to DecodeStructuredModeMessage.
e.g. 
I could do:
```c#
var evt = response.Content.ToDynamicJson.Value[0].Event;
```
But at that point I would need to somehow convert evt to bytes and I wasn't sure how to do this.

File: sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/samples/README.md [hunk 2]
```diff
@@ -13,3 +13,4 @@ description: Samples for the Azure.Messaging.EventGrid.Namespaces client library
 Before starting, take a look at the Azure Event Grid [README](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/README.md) for more information on how to create an Event Grid custom topic or domain using the Azure portal/Azure CLI, and retrieving the designated endpoint and credential.
 
 - [Using Namespace Topics](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/samples/Sample1_Namespaces.md)
+- [Using the Cloud Native CloudEvent type]
```
Comment (JoshLove-msft) [inline sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/samples/README.md:16]: Will need to add the URL once this merges

File: sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/Azure.Messaging.EventGrid.Namespaces.sln [hunk 3]
```diff
@@ -8,6 +8,13 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Azure.Messaging.EventGrid.N
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Azure.Messaging.EventGrid.Namespaces.Tests", "tests\Azure.Messaging.EventGrid.Namespaces.Tests.csproj", "{1F1CD1D4-9932-4B73-99D8-C252A67D4B46}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Samples", "Samples", "{5E3752BC-2C04-499F-B2C4-28426B3C344A}"
+	ProjectSection(SolutionItems) = preProject
```
Comment (jsquire) [inline sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/Azure.Messaging.EventGrid.Namespaces.sln:12]: This section looks odd.  Was including the markdown intentional here?

File: sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/samples/Sample2_CNCF.md (hunk 1 above)
Comment (jsquire) [reply sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/samples/Sample2_CNCF.md:26]: Maybe something like:

```csharp
var byteStuff = Encoding.UTF8.GetBytes(
    response.Content.ToDynamicJson().Value[0].Event.ToString());
```

So long as `Event` is of type `string`, I believe that would work.

**Review by jsquire — APPROVED** [review]

File: sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/Azure.Messaging.EventGrid.Namespaces.sln (hunk 3 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/Azure.Messaging.EventGrid.Namespaces.sln:12]: Yep, I did the same thing for Service Bus - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/Azure.Messaging.ServiceBus.sln#L22-L41

File: sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/samples/Sample2_CNCF.md (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid.Namespaces/samples/Sample2_CNCF.md:26]: Nice.. that works

**Pull request merged**